client/src/components/Workflow/Editor/Index.vue

```diff
@@ -653,7 +654,10 @@ export default {
         },
         onLabel(nodeId, newLabel) {
             const step = { ...this.steps[nodeId], label: newLabel };
+            const oldLabel = this.steps[nodeId].label;
             this.onUpdateStep(step);
+            const newMarkdown = replaceLabel(this.markdownText, "step", oldLabel, newLabel);
```

Comment:
The report functionality distinguishes between inputs and steps - steps are assumed to be things associated with jobs and inputs are "Input Dataset", "Input Dataset Collection", or "Input Parameters". Can you dispatch on that in here and then test it?

I think we should inform users - maybe via a Toast style message that their markdown report has been updated if it has changed? Also we probably want to catch exceptions coming out of replace label and warn users if invalid markdown is preventing the update but still let them make the label change to the workflow.

We also need to handle output labels but that can be a follow up PR for sure.

Reply:
I'll add the Toast message and some exceptions handling of course. Great idea and it really eases my stress over changing something the user has created. Thanks!

@jmchilton But regarding the inputs vs steps, this seems to be working for Inputs, as well as steps. Unless I'm testing it incorrectly. But when I select any of the Jobs objects to insert in the report (Job Parameters, Job Metrics, etc.) and choose a label I've associated with an Input Dataset/Colletction. Then change that label, the label is changing in the report.  Is there a different option that I need to select in the Markdown Editor, that doesn't say step="name_of_label"?

Reply:
```suggestion
            const stepType = this.steps[nodeId].type;
            const isInput = ["data_input", "data_collection_input", "parameter_input"].indexOf(stepType) >= 0;
            const labelType = isInput ? "input" : "step";
            const newMarkdown = replaceLabel(this.markdownText, labelType, oldLabel, newLabel);
```